Public/Src/Cache/ContentStore/Library/Utils/ResultsExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Synchronization;
+
+namespace BuildXL.Cache.ContentStore.Utils
+{
+    internal static class ResultsExtensions
+    {
+        /// <nodoc />
+        public static TResult WithLockAcquisitionDuration<TResult, TLockKey>(this TResult result, in LockSet<TLockKey>.LockHandle handle)
+            where TResult : ResultBase
+            where TLockKey : IEquatable<TLockKey>
+        {
+            if (handle.LockAcquisitionDuration != null)
+            {
+                var message = $", lock acquired by {(long)handle.LockAcquisitionDuration.Value.TotalMilliseconds}ms, ";
+                result.ExtraDiagnosticsMessage = message;

```

Comment:
Is this the only place you are using the new ExtraDiagnosticsMessage property? Wouldn it be better to just add a LockAcquisitionDuration property to Result?

Reply:
This is the only place, but this aspect is added into 3 different (and unrelated) result types: OpenStreamResult, PlaceFileResult and PutFIleResult.
My first idea was to add `LockDuration` property, but once I realized that the types are unrelated I decided to go with a more general approach and add an ability to put extra message into the result.